obsplus/events/get_events.py

```diff
@@ -11,23 +11,68 @@
 
 import obsplus
 
-UNSUPPORTED_PARAMS = {
-    "minradius",
-    "maxradius",
-    "latitude",
-    "longitude",
-    "magnitude_type",
-    "events",
-    "contributor",
-}
+CIRCULAR_PARAMS = {"latitude", "longitude", "minradius", "maxradius"}
+
+NONCIRCULAR_PARAMS = {"minlongitude", "maxlongitude", "minlatitude", "maxlatitude"}
+
+UNSUPPORTED_PARAMS = {"magnitude_type", "events", "contributor"}
 CLIENT_SUPPORTED = set(inspect.signature(Client.get_events).parameters)
 SUPPORTED_PARAMS = CLIENT_SUPPORTED - UNSUPPORTED_PARAMS
 
 
+def _sanitize_circular_search(**kwargs):
+    if CIRCULAR_PARAMS.intersection(kwargs):
+        if NONCIRCULAR_PARAMS.intersection(kwargs):
+            raise ValueError(
+                "{0} cannot be used with {1}".format(
+                    NONCIRCULAR_PARAMS.intersection(kwargs),
+                    CIRCULAR_PARAMS.intersection(kwargs),
+                )
+            )
+        if not {"latitude", "longitude"}.issubset(kwargs):
+            raise ValueError("Circular search requires both longitude and latitude")
+        # If neither minradius not maxradius they just want everything.
+        if "minradius" not in kwargs and "maxradius" not in kwargs:
+            _ = kwargs.pop("latitude", None)
+            _ = kwargs.pop("longitude", None)
+    # Split parameters that are supported on sql and those that are not.
+    circular_kwargs = {}
+    for key in CIRCULAR_PARAMS:
+        value = kwargs.pop(key, None)
+        if value:
+            circular_kwargs.update({key: value})
+    return circular_kwargs, kwargs
+
+
 def _get_ids(df, kwargs) -> set:
```

Comment:
If it will help you, feel free to add logic for using the `degrees` parameter to `get_events` which would be passed to `calculate_distance`. Obsplus' `get_events` interface can be a super set of obspy's (but try to make it clear that the degrees parameter may not be supported by other Clients in the get_events docstring).

Reply:
I'm not sure the best way to do that - just add a `degrees` kwarg to `get_events` and go from there, or add `degrees` to `SUPPORTED_PARAMS`?  Adding it would make it simpler to test the distance functions by essentially using the same test but with distances in m.